rgo/builders/copr.py

```diff
@@ -161,19 +164,44 @@ def build(self, component, with_build_id=None, after_build_id=None):
             "after_build_id": after_build_id,
         }
 
-        if self.chroots:
-            buildopts["chroots"] = self.chroots
-
-        build = self.client.build_proxy.create_from_file(self.owner, self.name, component.srpm, buildopts=buildopts)
-        LOGGER.info('Submitted Copr build for %s chroots: %s %s%sURL: %s' % (
-            component.name,
-            ", ".join(buildopts["chroots"]) if "chroots" in buildopts else "(all)",
-            "with: %s " % with_build_id if with_build_id is not None else "",
-            "after: %s " % after_build_id if after_build_id is not None else "",
-            self._build_url(build.id)
-        ))
+        if self.cmdline_input_chroots:
+            buildopts["chroots"] = self.cmdline_input_chroots
+
+        overriden_chroots = set()
+        for override in component.overrides:
+            overriden_chroots.update(override.roots)
+
+        # if we have atleast one default (not overriden) chroot do the default build
+        if (list(self.all_project_chroots - overriden_chroots)):
+            buildopts["chroots"] = list(self.all_project_chroots - overriden_chroots)
+            build = self.client.build_proxy.create_from_file(self.owner,
+                                                             self.name,
+                                                             component.srpm,
+                                                             buildopts=buildopts)
+            LOGGER.info('Submitted Copr build for %s chroots: %s %s%sURL: %s' % (
+                component.name, ", ".join(buildopts["chroots"]),
+                "with: %s " % with_build_id if with_build_id is not None else "",
+                "after: %s " % after_build_id if after_build_id is not None else "",
+                self._build_url(build.id)
+            ))
+            default_build_id = build.id
+
+        for override in component.overrides:
+            if override.roots:
+                buildopts["chroots"] = list(override.roots)
+                build = self.client.build_proxy.create_from_file(self.owner,
+                                                                 self.name,
+                                                                 override.srpm,
+                                                                 buildopts=buildopts)
+                LOGGER.info('Submitted Copr build for %s chroots: %s %s%sURL: %s' % (
+                    component.name, ", ".join(buildopts["chroots"]),
+                    "with: %s " % with_build_id if with_build_id is not None else "",
+                    "after: %s " % after_build_id if after_build_id is not None else "",
+                    self._build_url(build.id)
+                ))
+                override.build_id = build.id
```

Comment:
The code in this `for` loop is again very similar to the one called above for the non-overridden builds, it should go into a private method of this class. It's just a few calls, but the LOGGER line itself is quite non-trivial and shouldn't be duplicated.